services/auth/src/signaturevalidator.js

```diff
@@ -330,16 +332,23 @@ const createSignatureValidator = function(options) {
 
         credentials = authResult.credentials;
         attributes = authResult.attributes;
+      } else {
+        // No auth provided
+        result = {
+          status: 'no-auth',
+          scheme: 'none',
+          expires: new Date(Date.now() + 15 * 60 * 1000), // 15 minutes in future
+          scopes: options.expandScopes(['assume:anonymous']),
+        };
       }
-
-      result = {
+      result = result || {
         status: 'auth-success',
         scheme: 'hawk',
         expires: credentials.expires,
         scopes: credentials.scopes,
         clientId: credentials.clientId,
       };
-      if (attributes.hash) {
+      if (attributes && attributes.hash) {
```

Comment:
this isn't 100% identical to the old logic. It is probably fine to make this change but is there a reason we did?

Reply:
Without a check for `attributes`, it was generating the following error:

```
  services/auth/test/client_test.js (mock)
    ✓ ping
    ✓ auth.client (root credentials)
    1) auth.client (no credentials)


  2 passing (2s)
  18 pending
  1 failing

  1) services/auth/test/client_test.js (mock)
       auth.client (no credentials):
     Error: Cannot read property 'hash' of undefined

---

* method:     client
* errorCode:  AuthenticationFailed
* statusCode: 401
* time:       2020-10-21T20:17:47.023Z
```